signature/DeviceSignature.c

```diff
@@ -126,12 +128,22 @@ void getEui64(uint8_t eui[8], uint16_t offset) {
 
 int8_t sigInit(void)
 {
-	uint16_t offset = findSignature(0, 0);
+	uint16_t offset = findSignature(SIGNATURE_TYPE_EUI64, 0);
 	if (offset < 0xFFFF) {
 		mStatus = SIG_GOOD;
 		getEui64(mEUI64, offset);
 	}
 	else {
+		semver_t v = sigGetBoardVersion();
+		// If there is no EUI64 in the signature, signature still can be valid,
+		// check for board version and use built-in EUI.
+		if ((v.major > 0) || (v.minor > 0) || (v.patch > 0)) {
+			if (platform_eui(mEUI64)) {
```

Comment:
It would be better to initialize the mEUI64 even if the signature is completely empty. Mostly useful during testing. So I would propose you swap the checks, first init platform_eui, then check signature for version.